experimental/adaptive-dialog/docs/recognizers-rules-steps-reference.md

```diff
@@ -0,0 +1,739 @@
+            {
+                new Choice("Adaptive card"),
+                new Choice("Hero card"),
+                new Choice("Video card")
+            }
+        }, 
+        // Use SwitchCondition step to dispatch to right dialog based on choice input.
+        new SwitchCondition()
+        {
+            Condition = "turn.cardDialog.cardChoice",
+            Cases = new Dictionary<string, List<IDialog>>()
+            {
+                { "Adaptive card", new List<IDialog>() { new BeginDialog("ShowAdaptiveCardDialog")} },
+                { "Hero card", new List<IDialog>() { new BeginDialog("ShowHeroCardDialog")} },
+                { "Video card", new List<IDialog>() { new BeginDialog("ShowVideoCardDialog")} }
+            }
+        }
+}));
+```
+
+### EndTurn
+Ends the current turn of conversation.
+
+``` C#
+new EndTurn()
+```
+
+### BeginDialog
+Invoke and begin a new dialog. Begin dialog takes the target dialog's name and the target dialog can be any type of dialog including Adaptive dialog or Waterfall dialog etc.
+
+``` C#
+new BeginDialog("BookFlightDialog")
+```
+
+
+### EndDialog
+Ends the active dialog. 
+
+**Note:** Adaptive dialogs by default will end automatically if the dialog has run out of steps to execute. To override this behavior, simply set the `AutoEndDialog` property on Adaptive Dialog to false. 
+
+``` C#
+new EndDialog()
+```
+
+### CancelAllDialog
+Cancels all active dialogs including any active parent dialogs.
+
+``` C#
+new CancelAllDialog()
+```
+
+### ReplaceDialog
+Replace current dialog with a new dialog by name.
+
+``` C#
+// This sample illustrates use of ReplaceDialog tied to explict user confirmation 
+// to switch to a different dialog.
+
+// Create an adaptive dialog.
+var getUserName = new AdaptiveDialog("getUserName");
+getUserName.AddRule(new IntentRule("getUserName",
+steps: new List<IDialog>() {
+    new TextInput()
+    {
+        Property = "user.name",
+        Prompt = new ActivityTemplate("What is your name?")
+    },
+    new SendActivity("Hello {user.name}, nice to meet you!")
+}));
+
+getUserName.AddRule(new IntentRule("GetWeather",
+steps: new List<IDialog>()
+{
+    // confirm with user that they do want to switch to another dialog
+    new ChoiceInput()
+    {
+        Prompt = new ActivityTemplate("Are you sure you want to switch to talk about weather?"),
+        Property = "turn.contoso.getweather.confirmchoice",
+        Choices = new List<Choice>()
+        {
+            new Choice("Yes"),
+            new Choice("No")
+        }
+    },
+    new SwitchCondition()
+    {
+        Condition = "turn.contoso.getweather.confirmchoice",
+        Cases = new Dictionary<String, List<IDialog>>()
+        {
+            // Call ReplaceDialog to switch to a different dialog. 
+            // BeginDialog will keep current dialog in the stack to be resumed after child dialog ends.
+            // ReplaceDialog will remove current dialog from the stack and add the new dialog.
```

Comment:
This is important compare to BeginDialog. Should we update both BeginDialog and ReplaceDialog with relevnet info/ background?